File: app/src/main/java/com/gmail/mostafa/ma/saleh/yawa/utilities/DialogUtils.java
package com.gmail.mostafa.ma.saleh.yawa.utilities;

import android.app.Dialog;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.support.annotation.NonNull;
import android.view.Window;
import android.widget.ProgressBar;

/**
 * Created by Mostafa on 11/11/2017.
 */

public class DialogUtils {
    public static Dialog showProgressDialog(@NonNull Context context) {
        Dialog progressDialog = new Dialog(context);
        progressDialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        progressDialog.setContentView(new ProgressBar(context));
        //noinspection ConstantConditions
        progressDialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        progressDialog.setCanceledOnTouchOutside(false);
        progressDialog.setCancelable(false);
        progressDialog.show();
        return progressDialog;
    }
}

File: app/src/main/java/com/gmail/mostafa/ma/saleh/yawa/utilities/DialogUtils.kt
package com.gmail.mostafa.ma.saleh.yawa.utilities

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.view.Window
import android.widget.ProgressBar

/**
 * Created by Mostafa on 11/11/2017.
 */

object DialogUtils {
    fun showProgressDialog(context: Context): Dialog {
        Dialog(context).run {
            requestWindowFeature(Window.FEATURE_NO_TITLE)
            setContentView(ProgressBar(context))
            window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            setCanceledOnTouchOutside(false)
            setCancelable(false)
            show()
            return this
        }
    }

    fun showListDialog(context: Context, items: List<String>, onClickListener: DialogInterface.OnClickListener) {
        AlertDialog.Builder(context)
                .setSingleChoiceItems(items.toTypedArray(), -1, onClickListener)
                .show()
    }
}
